docs/source/advanced/rag.rst

```diff
@@ -0,0 +1,614 @@
+WebQSPDataset.
+
+.. code:: python
+
+    from torch_geometric.datasets import WebQSPDataset
+
+    num_questions = 100
+    ds = WebQSPDataset('small_sample', limit=num_questions)
+
+
+WebQSP is a dataset that is based off of a subset of the Freebase
+Knowledge Graph, which is an open-source knowledge graph formerly
+maintained by Google. For each question-answer pair in the dataset, a
+subgraph was chosen based on a Semantic SPARQL search on the larger
+knowledge graph, to provide relevent context on finding the answer. So
+each entry in the dataset consists of:
+- A question to be answered
+- The answer
+- A knowledge graph subgraph of Freebase that has the context
+needed to answer the question.
+
+.. code:: python
+
+    ds.raw_dataset
+
+    >>> Dataset({
+        features: ['id', 'question', 'answer', 'q_entity', 'a_entity', 'graph', 'choices'],
+        num_rows: 100
+    })
+
+
+
+.. code:: python
+
+    ds.raw_dataset[0]
+
+
+    >>> {'id': 'WebQTrn-0',
+     'question': 'what is the name of justin bieber brother',
+     'answer': ['Jaxon Bieber'],
+     'q_entity': ['Justin Bieber'],
+     'a_entity': ['Jaxon Bieber'],
+     'graph': [['P!nk', 'freebase.valuenotation.is_reviewed', 'Gender'],
+      ['1Club.FM: Power', 'broadcast.content.artist', 'P!nk'],
+      ...],
+     'choices': []}
+
+
+
+Although this dataset can be trained on as-is, a couple problems emerge
+from doing so:
+1. A retrieval algorithm needs to be implemented and
+executed during inference time, that might not appropriately correspond
+to the algorithm that was used to generate the dataset subgraphs.
+1. The dataset as is not stored computationally efficiently, as there will
+exist many duplicate nodes and edges that are shared between the
+questions.
+
+As a result, it makes sense in this scenario to be able to encode all
+the entries into a large knowledge graph, so that duplicate nodes and
+edges can be avoided, and so that alternative retrieval algorithms can
+be tried. We can do this with the LargeGraphIndexer class:
+
+.. code:: python
+
+    from torch_geometric.data import LargeGraphIndexer, Data, get_features_for_triplets_groups
+    from torch_geometric.nn.nlp import SentenceTransformer
+    import time
+    import torch
+    import tqdm
+    from itertools import chain
+    import networkx as nx
+
+.. code:: python
+
+    raw_dataset_graphs = [[tuple(trip) for trip in graph] for graph in ds.raw_dataset['graph']]
+    print(raw_dataset_graphs[0][:10])
+
+    >>> [('P!nk', 'freebase.valuenotation.is_reviewed', 'Gender'), ('1Club.FM: Power', 'broadcast.content.artist', 'P!nk'), ...]
+
+
+To show the benefits of this indexer in action, we will use the
+following model to encode this sample of graphs using LargeGraphIndexer,
+along with naively.
+
+.. code:: python
+
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    model = SentenceTransformer(model_name='sentence-transformers/all-roberta-large-v1').to(device)
```

Comment:
whats stale about it? regardless its part of rag.rst. the whole file should be moved to the future webqsp PR anywhere since it relies on those changes